fibonacci/ocaml/code.ml

```diff
@@ -0,0 +1,17 @@
+
+
+let fibonacci n =
+  let rec aux a b n =
+    if n = 0 then a
+    else aux b (a + b) (n - 1)
+  in
```

Comment:
I don't know Ocaml. Does this match the other naive implementation from the other languages?

Reply:
I do not use functional languages that much but this program looks exactly like if someone asked me to implement Fibonacci series with naive approach. The only difference is that we go from bottom to top but I cannot imagine how to do it the other way using functional paradigm

Reply:
This is a recursive implementation the same as the other but with a difference. It's tail call recursion. So OCaml compiler can optimize in an extreme way such type or recursion. 

So, it's recursion, no caching is done, and the good performance comes from the use of tail calls.

But for the sake of comparison, I'll change it to "non tail call"recursion.

Reply:
Changed. It's important to note that tail call recursion is always preferred (I guess in any language) in OCaml.